src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -605,52 +514,40 @@ protected internal override bool TryComputeLength(out long length)
 
         protected override void Dispose(bool disposing)
         {
-            _fetchResponse.Dispose();
+            _controller.Dispose();
             base.Dispose(disposing);
         }
     }
 
-    internal sealed class WasmHttpReadStream : Stream
+    internal sealed class BrowserHttpReadStream : Stream
     {
-        private WasmFetchResponse _fetchResponse;
+        private BrowserHttpController _controller; // we own the object and have to dispose it
 
-        public WasmHttpReadStream(WasmFetchResponse fetchResponse)
+        public BrowserHttpReadStream(BrowserHttpController fetchResponse)
         {
-            _fetchResponse = fetchResponse;
+            _controller = fetchResponse;
         }
 
-        public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken)
+        public override ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken)
         {
             ArgumentNullException.ThrowIfNull(buffer, nameof(buffer));
-            _fetchResponse.ThrowIfDisposed();
-#if FEATURE_WASM_THREADS
-            return await _fetchResponse.FetchResponse!.SynchronizationContext.Post(() => Impl(this, buffer, cancellationToken)).ConfigureAwait(true);
-#else
-            return await Impl(this, buffer, cancellationToken).ConfigureAwait(true);
-#endif
+            _controller.ThrowIfDisposed();
 
-            static async Task<int> Impl(WasmHttpReadStream self, Memory<byte> buffer, CancellationToken cancellationToken)
+            MemoryHandle pinBuffer = buffer.Pin();
+            try
             {
-                self._fetchResponse.ThrowIfDisposed();
-                Task<int> promise;
-                using (Buffers.MemoryHandle handle = buffer.Pin())
-                {
-#if FEATURE_WASM_THREADS
-                    lock (self._fetchResponse.ThisLock)
-                    {
-#endif
-                        self._fetchResponse.ThrowIfDisposed();
-                        promise = GetStreamedResponseBytesUnsafe(self._fetchResponse, buffer, handle);
-#if FEATURE_WASM_THREADS
-                    } //lock
-#endif
-                    int response = await BrowserHttpInterop.CancelationHelper(promise, cancellationToken, self._fetchResponse.FetchResponse).ConfigureAwait(true);
-                    return response;
-                }
+                _controller.ThrowIfDisposed();
```

Comment:
Unrelated to this PR but isn't one `ThrowIfDisposed()` (e.g. the one above) enough here?